fastai/optimizer.py

```diff
@@ -27,15 +38,27 @@ def freeze(self):
         assert(len(self.param_lists)>1)
         self.freeze_to(-1)
 
-    def set_freeze(self, n, rg, ignore_force_train=False):
+    def set_freeze(self,
+        n: int, # layer `n` to un/freeze
+        rg: bool, # requires gradient
+        ignore_force_train=False  # overwrites "force_train" or batch norm always train even if freezed
+    ):
         for p in self.param_lists[n]: p.requires_grad_(rg or (state.get('force_train', False) and not ignore_force_train))
 
     def unfreeze(self): self.freeze_to(0)
-    def set_hypers(self, **kwargs): L(kwargs.items()).starmap(self.set_hyper)
-    def _set_hyper(self, k, v):
+    def set_hypers(self,
+        **kwargs  # list of hyper parameters to set with shape `(requires gradient, parameters, param group, state, hyper parameter)`
+    ): L(kwargs.items()).starmap(self.set_hyper)
```

Comment:
Can we use `delegates` to propagate this at all?

Reply:
mmm, I guess `@delegates(set_hyper)` will do?